### PR TITLE
DF-1624 - handle approved from rejected JobAds

### PR DESCRIPTION
**Pull request by cjpsycos**

### CODE REVIEW

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisement.java [hunk 1]
```diff
@@ -313,6 +277,12 @@ public void approve(String stellennummerAvam, LocalDate date, boolean reportingO
         this.stellennummerAvam = Condition.notBlank(stellennummerAvam);
         this.approvalDate = Condition.notNull(date);
 
+        this.rejectionCode = null;
+        this.rejectionReason = null;
+        this.rejectionDate = null;
+        this.cancellationCode = null;
```
Comment (birom) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisement.java]: Why do we need to reset the ``cancellationCode`` and ``cancellationDate`` ?
According to the ``JobAdvertisementStatus`` enum the ``CANCELLED`` is a terminal state:
```
CANCELLED.allowedDestinationStates = new JobAdvertisementStatus[]{CANCELLED};
```

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/GeoPoint.java [hunk 2]
```diff
@@ -27,4 +28,18 @@ public double getLon() {
...
+        if (o == null || getClass() != o.getClass()) return false;
+        GeoPoint geoPoint = (GeoPoint) o;
+        return Double.compare(geoPoint.lon, lon) == 0 &&
+                Double.compare(geoPoint.lat, lat) == 0;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(lon, lat);
+    }
```
Comment (tonim42) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/GeoPoint.java]: Das hat keinen Bezug zu 1624 oder?

Comment (cjpsycos) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/GeoPoint.java]: nein. haben nur ein gefundenen fehler gefixt (boy scout rule)

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisement.java (hunk 1 above)
Comment (fabiangotzen) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisement.java]: Yes you are right, strictly speaking it should never be the case that we need to null these 2 values. The only reason we added it was to make it explicit that an ApprovalDto can and should not have any rejection or cancellation-related data.

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/GeoPoint.java (hunk 2 above)
Comment (tonim42) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/GeoPoint.java]: Jetzt verstehe ich - das ist der Fix zu 1264 :)

**Pull request merged**